src/main/java/net/glowstone/entity/projectile/GlowArrow.java

```diff
@@ -0,0 +1,130 @@
+package net.glowstone.entity.projectile;
+
+import lombok.Getter;
+import lombok.Setter;
+import net.glowstone.block.blocktype.BlockTnt;
+import net.glowstone.entity.meta.MetadataIndex;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.entity.Arrow;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Monster;
+import org.bukkit.event.entity.EntityDamageEvent;
+import org.bukkit.projectiles.ProjectileSource;
+import org.bukkit.util.Vector;
+
+public class GlowArrow extends GlowProjectile implements Arrow {
+
+    private volatile PickupStatus customPickupStatus = null;
+    private final Arrow.Spigot spigot = new GlowArrow.Spigot();
+    @Getter
+    @Setter
+    private int knockbackStrength;
+    /**
+     * How long an entity burns after being shot with a burning arrow.
+     */
+    private static final int TARGET_BURN_TICKS = 100;
+
+    /**
+     * Creates an arrow entity.
+     *
+     * @param location the initial location
+     */
+    public GlowArrow(Location location) {
+        super(location);
+        setGravityAccel(new Vector(0, -0.05, 0));
+        setBoundingBox(0.5, 0.5);
+    }
+
+    @Override
+    protected void pulsePhysics() {
+        super.pulsePhysics();
+        if (!isOnGround()) {
+            location.add(getVelocity());
+            if (isTouchingMaterial(Material.WATER)
+                    || isTouchingMaterial(Material.STATIONARY_WATER)) {
+                setFireTicks(0);
+            } else if (isTouchingMaterial(Material.LAVA)
+                    || isTouchingMaterial(Material.STATIONARY_LAVA)) {
+                setFireTicks(Integer.MAX_VALUE);
+            }
+        }
+    }
+
+    @Override
+    public void setOnGround(boolean onGround) {
+        super.setOnGround(onGround);
+        if (onGround) {
+            setVelocity(new Vector(0, 0, 0));
+        }
+    }
+
+    @Override
+    public void collide(Block block) {
+        setFireTicks(0); // Arrows stop burning when they land, and ignite only TNT
+        if (block.getType() == Material.TNT) {
+            BlockTnt.igniteBlock(block, false);
+        }
+        setOnGround(true);
+        // Don't despawn
```

Comment:
Arrows should be despawned eventually, perhaps with a configurable value like in Spigot/Paper.

Reply:
Added TODO for this, while I find out the best way to schedule the despawn event.